### PR TITLE
refactor: add dateTimeObject and dateTimeString to Cues for backwards compatibility

### PR DESCRIPTION
**Pull request by harisha-swaminathan**
Adds `dateTimeObject` and `dateTimeString` back to segment metadata cues for backwards compatibility

### CODE REVIEW

File: src/util/time.js [hunk 1]
```diff
@@ -74,15 +74,15 @@ export const findSegmentForProgramTime = (programTime, playlist) => {
 
   let segment = playlist.segments[0];
 
-  if (dateTimeObject < segment.programDateTime) {
+  if (dateTimeObject < new Date(segment.programDateTime)) {
```
Comment (wseymour15) [inline src/util/time.js:77]: I'm assuming this is unrelated to adding back the old properties. Just for my understanding, what is the purpose for these changes? I am sure there is a good reason :)

Comment (harisha-swaminathan) [reply src/util/time.js:77]: So this is related to the changes I made in the original PR that replaced dateTimeObject/String with `programDateTime` here https://github.com/videojs/http-streaming/pull/1407/files#diff-ba19ddab36415030a61221adf95d07ad0ec76ce6453cdd837b69285eb4533974L77

With the changes I made in that PR, we were comparing `dateTimeObject` (a date object)  and `programDateTime` (a timestamp represented in ms) and it works fine but just wanted them both to be of the same type just to be safe.

Comment (wseymour15) [reply src/util/time.js:77]: Makes sense, thanks!

**Review by wseymour15 — APPROVED** [review]
